mapper-processor/src/main/java/com/datastax/oss/driver/internal/mapper/processor/dao/EntityInjectorGenerator.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.driver.internal.mapper.processor.dao;
+
+import com.datastax.oss.driver.api.mapper.annotations.Entity;
+import com.datastax.oss.driver.api.mapper.annotations.EntityInjector;
+import com.datastax.oss.driver.internal.mapper.processor.PartialClassGenerator;
+import com.datastax.oss.driver.internal.mapper.processor.ProcessorContext;
+import com.datastax.oss.driver.internal.mapper.processor.SkipGenerationException;
+import com.squareup.javapoet.ClassName;
+import com.squareup.javapoet.MethodSpec;
+import com.squareup.javapoet.TypeSpec;
+import javax.lang.model.element.Element;
+import javax.lang.model.element.ElementKind;
+import javax.lang.model.element.ExecutableElement;
+import javax.lang.model.element.Modifier;
+import javax.lang.model.element.TypeElement;
+import javax.lang.model.element.VariableElement;
+import javax.lang.model.type.DeclaredType;
+import javax.lang.model.type.TypeKind;
+import javax.lang.model.type.TypeMirror;
+
+public class EntityInjectorGenerator implements PartialClassGenerator {
+
+  private final ExecutableElement methodElement;
+  private final DaoImplementationGenerator daoImplementationGenerator;
+  private final String entityParameterName;
+  private final TypeElement entityElement;
+  private final String targetParameterName;
+
+  public EntityInjectorGenerator(
+      ExecutableElement methodElement,
+      DaoImplementationGenerator daoImplementationGenerator,
+      ProcessorContext context) {
+    this.methodElement = methodElement;
+    this.daoImplementationGenerator = daoImplementationGenerator;
+    // We're expecting a void method, where one parameter is an annotated entity, and the other a
+    // SettableByName or subtype
+    TypeMirror returnType = methodElement.getReturnType();
+    if (returnType.getKind() != TypeKind.VOID || methodElement.getParameters().size() != 2) {
```

Comment:
maybe we can name that condition like `isVoidMethodAndTwoHasParameters` and extract such method?
Also, I don't see a check for:
>one parameter is an annotated entity, and the other a
>    // SettableByName or subtype

Only validation on a number of arguments. I see that it is checked after, maybe we can do a pre-validation or all pre-conditions needed at the entry of this processing?

Reply:
As we implement more query methods we can extract common patterns if we see them emerge. Right now I'm not sure if `isVoidMethodAndTwoHasParameters` is going to be useful anywhere else.

The check for `SettableByName` is at line 65:
```java
      if (context.getClassUtils().implementsSettableByName(parameterType)) {
```